lms/services/upsert.py

```diff
@@ -55,4 +60,11 @@ def bulk_upsert(
     # back
     mark_changed(db)
 
-    return result
+    if not returning:
+        return result
+
+    return (
+        db.query(model_class)
+        .filter(tuple_(*index_elements_columns).in_(result.all()))
+        .all()
+    )
```

Comment:
What's the difference between these two returns? Would it be possible to make it do `returning=True` all the time and avoid the extra parameter?

Reply:
We don't use the current return value of this function (and in fact, this PR changes it even when returning=False)

My only problem with always returning the last part is that it will add and extra query + object serialization to for example the current `File.upsert`.

Reply:
Changed to return the query in all cases (with out the call to `.all()`) :+1: